src/main.ts

```diff
@@ -284,36 +301,70 @@ function getFetchInfo(ref: string, commit: string): IFetchInfo {
   return result
 }
 
-async function configGitAuth(token: string) {
+async function configGitAuthForSubmodules(token: string, repoDir: string) {
+  // Set authentication
+  const basicCredential = Buffer.from(
+    `x-access-token:${token}`,
+    'utf8'
+  ).toString('base64')
+  core.setSecret(basicCredential)
+
+  await exec.exec(
+    `git submodule foreach --recursive sh -c "git config --local --add 'http.https://github.com/.extraheader' 'AUTHORIZATION: basic ${basicCredential}'"`,
+    [],
+    { cwd: repoDir }
+  )
+  await exec.exec(
+    `git submodule foreach --recursive sh -c "git config --local --add 'url.https://github.com/.insteadOf' 'git@github.com:'"`,
+    [],
```

Comment:
Should we also be doing ignoreReturnCode: true here as we do in configGitAuth ?

Reply:
We only ignore the result in cleanup of previous authentication header. But if command that sets the header fails, we want to fail the action as well.